src/boot/efi/meson.build

```diff
@@ -39,312 +142,206 @@ systemd_boot_sources = '''
-                        error('gnu-efi support requested, but cannot find efi.lds')
-                else
-                        have_gnu_efi = false
-                endif
+        pkgver = get_option('sbat-distro-version')
+        if pkgver == ''
+                efi_conf.set('SBAT_DISTRO_VERSION', 'GIT_VERSION')
+        else
+                efi_conf.set_quoted('SBAT_DISTRO_VERSION', pkgver)
         endif
 endif
 
-if have_gnu_efi
-        compile_args = cc.get_supported_arguments(
-                basic_disabled_warnings +
-                possible_common_cc_flags + [
-                        '-fno-stack-protector',
-                        '-fno-strict-aliasing',
-                        '-fpic',
-                        '-fwide-exec-charset=UCS2',
-                        '-Wall',
-                        '-Wextra',
-                        '-Wsign-compare',
-                ]
-        ) + [
-                '-nostdlib',
-                '-std=gnu99',
-                '-ffreestanding',
-                '-fshort-wchar',
-                '-isystem', efi_incdir,
-                '-isystem', efi_incdir / gnu_efi_path_arch,
-                '-I', fundamental_path,
-                '-DSD_BOOT',
-                '-DGNU_EFI_USE_MS_ABI',
-                '-include', efi_config_h,
-                '-include', version_h,
+efi_config_h = configure_file(
+        output : 'efi_config.h',
+        configuration : efi_conf)
+
+compile_args = cc.get_supported_arguments(
+        basic_disabled_warnings +
+        possible_common_cc_flags + [
+                '-fno-stack-protector',
+                '-fno-strict-aliasing',
+                '-fpic',
+                '-fwide-exec-charset=UCS2',
+                '-Wall',
+                '-Wextra',
+                '-Wsign-compare',
         ]
+) + [
+        '-nostdlib',
+        '-std=gnu99',
+        '-ffreestanding',
+        '-fshort-wchar',
+        '-isystem', efi_incdir,
+        '-isystem', efi_incdir / efi_arch[1],
+        '-I', fundamental_path,
+        '-DSD_BOOT',
+        '-DGNU_EFI_USE_MS_ABI',
+        '-include', efi_config_h,
+        '-include', version_h,
+]
+
+compile_args += cc.get_supported_arguments({
+        'ia32':   ['-mno-sse', '-mno-mmx'],
+        'x86_64': ['-mno-red-zone', '-mno-sse', '-mno-mmx'],
+        'arm':    ['-mgeneral-regs-only', '-mfpu=none'],
+}.get(efi_arch[1], []))
+
+# We are putting the efi_cc command line together ourselves, so make sure to pull any
+# relevant compiler flags from meson/CFLAGS as povided by the user or distro.
+
+if get_option('werror')
+        compile_args += ['-Werror']
+endif
+if get_option('debug')
+        compile_args += ['-ggdb', '-DEFI_DEBUG']
+endif
+if get_option('optimization') != '0'
+        compile_args += ['-O' + get_option('optimization')]
+endif
+if get_option('b_ndebug') == 'true' or (
+        get_option('b_ndebug') == 'if-release' and get_option('buildtype') in ['plain', 'release'])
```

Comment:
This change in indentation is not an improvement. Please avoid having the last line of the conditional indented the same as the block below.